Source/DafnyCore/Verifier/BoogieGenerator.Methods.cs

```diff
@@ -1263,9 +1263,7 @@ public partial class BoogieGenerator {
         reveal = new Boogie.IdentifierExpr(f.tok, revealVar);
         argsJF.Add(reveal);
       } else if (overridingFunction.IsOpaque || overridingFunction.IsMadeImplicitlyOpaque(options)) {
-        // We can't use a bound variable $fuel, because then one of the triggers won't be mentioning this $fuel.
-        // Instead, we do the next best thing: use the literal false.
-        reveal = new Boogie.LiteralExpr(f.tok, false);
+        reveal = GetRevealConstant(overridingFunction);
```

Comment:
This sounds largely plausible to be the cause of the issue. Thanks for spotting!